application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -306,7 +306,7 @@ private void handleActionWithMessage(@NotNull TagStatus requiredTagStatus,
      * @param id the id of the tag to get its content
      * @return the content of the tag, or {@code "Unable to retrieve content"} if was unable to
      */
-    private @NotNull String getTagContent(@NotNull Subcommand subcommand, @NotNull String id) {
+    private @Nullable String getTagContent(@NotNull Subcommand subcommand, @NotNull String id) {
```

Comment:
dont return null out of methods. use Optional instead.

That said, the call-site now also has to do the `Unable to retrieve content` part, no? Otherwise the UX will be ugly.

Reply:
Or is the last line supposed to never be executed? Then it should be `throw new AssertionError();`.

Reply:
Look at 7a812b783ce9d6569deed2c485fe38650360755e please